spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -101,6 +109,19 @@ public long getTaskExecutionCount(String taskName) {
 		return new ArrayList<String>(result);
 	}
 
+	@Override
+	public Page<TaskExecution> findAll(Pageable pageable) {
+		Set<TaskExecution> sortedSet = getTaskExecutionTreeSet();
+		sortedSet.addAll(taskExecutions.values());
+		List<TaskExecution> result = new ArrayList<>(sortedSet);
```

Comment:
In this case the key is the executionId.  And we can't use the value of the Map in the Comparator. :-(